sylvia-derive/src/querier.rs

```diff
@@ -173,3 +176,95 @@ where
         }
     }
 }
+
+pub struct ContractQuerier<'a> {
+    source: &'a ItemImpl,
+    variants: MsgVariants<'a, GenericParam>,
+    interfaces: &'a Interfaces,
+}
+
+impl<'a> ContractQuerier<'a> {
+    pub fn new(source: &'a ItemImpl, interfaces: &'a Interfaces) -> Self {
+        let variants = MsgVariants::new(source.as_variants(), MsgType::Query, &[], &None);
+        Self {
+            source,
+            variants,
+            interfaces,
+        }
+    }
+
+    pub fn emit(&self) -> TokenStream {
+        let sylvia = crate_module();
+        let Self {
+            source,
+            variants,
+            interfaces,
+        } = self;
+
+        let where_clause = &source.generics.where_clause;
+        let generics: Vec<_> = source.generics.params.iter().collect();
+        let contract = &source.self_ty;
+
+        let accessor = MsgType::Query.as_accessor_name(false);
```

Comment:
`as_accessor_name` and `emit_msg_name` hide some logic behind the bool flag argument, which is not totally clear when calling them. Also `as_accessor_name` returns Option, but it never produces `None`. Consider something like this for e.g.:
```
pub fn as_accessor_name(&self) -> Type {
    match self {
        MsgType::Instantiate => parse_quote! { Instantiate },
        MsgType::Exec => parse_quote! { Exec },
        MsgType::Query => parse_quote! { Query },
        MsgType::Migrate => parse_quote! { Migrate },
        MsgType::Sudo => parse_quote! { Sudo },
        MsgType::Reply => parse_quote! { Reply },
    }
}

pub fn as_accessor_wrapper_name(&self) -> Type {
    match self {
        MsgType::Exec if is_wrapper => parse_quote! { ContractExec },
        MsgType::Query if is_wrapper => parse_quote! { ContractQuery },
        _ => self.as_accessor_name()
    }
}
```

Reply:
This advice can be applied to other such methods that takes bools as params

Reply:
This `Option` I think is a remainder from time when we didn't support `sudo`/`reply` and should be removed.

I see a benefit of this approach. Since it's not directly related to this change I will do it in a separate PR in which I will do some post feature cleanup.

Reply:
Done in #297